unix/x0vncserver/XDesktop.cxx

```diff
@@ -753,6 +772,33 @@ bool XDesktop::handleGlobalEvent(XEvent* ev) {
 
     return true;
 #endif
+#ifdef HAVE_XFIXES
+  } else if (ev->type == EnterNotify) {
+    XCrossingEvent* cev;
+
+    if (!running)
+      return true;
+
+    cev = (XCrossingEvent*)ev;
+
+    if (cev->window != cev->root)
+      return false;
+
+    return setCursor();
+  } else if (ev->type == LeaveNotify) {
+    XCrossingEvent* cev;
+
+    if (!running)
+      return true;
+
+    cev = (XCrossingEvent*)ev;
+
+    if (cev->window == cev->root)
+      return false;
```

Comment:
It also doesn't seem to trigger when intended as I don't get a blank cursor when leaving the screen.

Reply:
Could this be because of Xephyr? I have never used it so I don't know how it works. I tested this one more time, adding debug messages into both `EnterNotify` and `LeaveNotify` as well as debug messages after the check so I can see whether it makes the cursor visible or whether it gets hidden and it does what I expect:
```
Thu Sep  1 08:09:17 2022
 XDesktop:    EnterNotify
 XDesktop:    Showing the cursor

Thu Sep  1 08:09:18 2022
 XDesktop:    LeaveNotify
 XDesktop:    Hiding the cursor
```
The `EnterNotify` is called correctly when moving back to the screen I'm connected to and same with the `LeaveNotify`.

Reply:
I doubt it is Xephyr, at least not all of the symptoms. Are you testing with a full desktop environment? Or with a minimal case? A key issue might be if the root window is actually visible, or if it is obscured by a "desktop" window.

In my case, I have just xterm and metacity running.

Reply:
I test this with Motif Window Manager (MWM), but now I also tried Metacity, using just `metacity` instead of the `mwm` command in the configuration I provided below and even though I couldn't do anything (e.g. type into xterm), it was still working for me as expected.

Reply:
I will try with Xephyr.

Reply:
Ok, I have tested Xephyr, but I don't think it can be used to verify this change. With Xephyr you have two separate X sessions, while with zaphod mode I think they are supposed to act like one. That's the reason why there was an issue as x0vncserver was getting events for the other screen.